diff --git a/node_modules/pm2/paths.js b/node_modules/pm2/paths.js
index b1e8382..f8d05fe 100644
--- a/node_modules/pm2/paths.js
+++ b/node_modules/pm2/paths.js
@@ -7,6 +7,7 @@
 var debug = require('debug')('pm2:paths');
 var p     = require('path');
 var fs    = require('fs')
+var os  = require("os")
 
 function getDefaultPM2Home() {
   var PM2_ROOT_PATH;
@@ -81,10 +82,11 @@ module.exports = function(PM2_HOME) {
 
   if (process.platform === 'win32' ||
       process.platform === 'win64') {
+       var USERNAME = os.userInfo().username
     //@todo instead of static unique rpc/pub file custom with PM2_HOME or UID
-    pm2_file_stucture.DAEMON_RPC_PORT = '\\\\.\\pipe\\rpc.sock';
-    pm2_file_stucture.DAEMON_PUB_PORT = '\\\\.\\pipe\\pub.sock';
-    pm2_file_stucture.INTERACTOR_RPC_PORT = '\\\\.\\pipe\\interactor.sock';
+    pm2_file_stucture.DAEMON_RPC_PORT = `\\\\.\\pipe\\${USERNAME}\\rpc.sock`;
+    pm2_file_stucture.DAEMON_PUB_PORT = `\\\\.\\pipe\\${USERNAME}\\pub.sock`;
+    pm2_file_stucture.INTERACTOR_RPC_PORT = `\\\\.\\pipe\\${USERNAME}\\interactor.sock`;
   }
 
   return pm2_file_stucture;
